src/stdlib_hashmap_open.f90

```diff
@@ -0,0 +1,875 @@
+            call expand_slots(map)
+        end if
+        map % call_count = map % call_count  + 1
+        base_slot = fibonacci_hash( hash_val, map % nbits )
+
+        offset = 0
+        PROBE_SUCCESSIVE_SLOTS: do
+            map % probe_count = map % probe_count + 1
+            test_slot = iand( base_slot + offset, map % index_mask )
+            inmap = map % slots(test_slot)
+            if ( inmap == 0 ) then
+                call allocate_open_map_entry(map, new_ent)
+                new_ent % hash_val = hash_val
+                call copy_key( key, new_ent % key )
+                if ( present( other ) ) &
+                    call copy_other( other, new_ent % other )
+                inmap = new_ent % inmap
+                map % inverse( inmap ) % target => new_ent
+                map % slots( test_slot ) = inmap
+                if ( present(conflict) ) conflict = .false.
+                return
+            else if ( inmap < 0 .or. &
+                inmap > map % num_entries + map % num_free ) then
+                error stop submodule_name // ' % ' // procedure // ': ' // &
+                    invalid_inmap
+            else if (.not. associated( map % inverse(inmap) % target ) ) then
+                error stop submodule_name // ' % ' // procedure // ': ' // &
+                    invalid_inmap
+            else
+                associate( target => map % inverse(inmap) % target )
+                  if ( hash_val == target % hash_val ) then
+                      if ( key == target % key ) then
+                          ! entry already exists
+                          if ( present(conflict) ) then
+                              conflict = .true.
+                          else
+                              error stop submodule_name // ' % ' // procedure &
+                                  // ': ' // conflicting_key
+                          end if
+                          return
+                      end if
+                  end if
+                end associate
+            end if
+            offset = offset + 1
+        end do PROBE_SUCCESSIVE_SLOTS
+
+    contains
+
+        subroutine allocate_open_map_entry(map, bucket)
+!         allocates a hash bucket
+            type(open_hashmap_type), intent(inout) :: map
+
+            type(open_map_entry_type), pointer, intent(out) :: bucket
+            type(open_map_entry_list), pointer :: free_list
+            type(open_map_entry_pool), pointer :: pool
+
+            pool => map % cache
+            map % num_entries = map % num_entries + 1
+            if ( associated(map % free_list) ) then
+!             Get hash bucket from free_list
+                free_list => map % free_list
+                bucket => free_list % target
+                map % free_list => free_list % next
+                free_list % target => null()
+                free_list % next => null()
+                if (bucket % inmap == 0) stop "bucket % inmap == 0"
```

Comment:
Unhandled fatal error `stop` will terminate the program, should this really be fatal?.

Reply:
If I have coded correctly the only way you can get `bucket % inmap == 0` is if there has been bit errors in either the address of `bucket`, the pointer to `inmap`, or the value of `inmap`.